torchao/quantization/prototype/mixed_precision/scripts/BO_acc.py

```diff
@@ -0,0 +1,419 @@
+        parameters=parameters_list,
+        name="test_quantize_BO",
+        objectives={"cal_PPL": ObjectiveProperties(minimize=True)},
+        choose_generation_strategy_kwargs={
+            "num_initialization_trials": num_initial, # the number of trials to build generation strategy
+        },
+        outcome_constraints=[constraint],
+    )
+
+    gpu_list = [int(i) for i in gpu_list.split(",")]
+
+    history=[]
+    trial_id = 0
+
+    # Set the multiprocessing start method to 'spawn'
+    mp.set_start_method("spawn", force=True)
+
+    # add initial points into the BO trials
+    for id in range(num_initial//len(gpu_list)):
+        processes = []
+        manager = mp.Manager()
+        return_dict = manager.dict()
+
+        # Start the worker processes
+        for i, gpu_id in enumerate(gpu_list):
+            ax_client.attach_trial(parameters=dict(initial_points_set[id*len(gpu_list)+i]))
+            p = mp.Process(target=eval_in_parallel, args=(gpu_id, checkpoint, limit, initial_points_set[id*len(gpu_list)+i], return_dict, i, trial_id))
+            trial_id += 1
+            p.start()
+            processes.append(p)
+
+        # Wait for all processes to finish
+        for p in processes:
+            p.join()
+
+        # Print the results after all processes have finished
+        print(return_dict)
+        for i in range(len(gpu_list)):
+            current_trial_id, config, eval_results = return_dict[i]
+            history.append((eval_results, config))
+            ax_client.complete_trial(trial_index=current_trial_id,raw_data=eval_results,)
+
+    # run new BO trials
+    for id in range(num_trials//len(gpu_list)):
+        processes = []
+        manager = mp.Manager()
+        return_dict = manager.dict()
+
+        # Start the worker processes
+        for i, gpu_id in enumerate(gpu_list):
+            parameters, trial_idx = ax_client.get_next_trial()
+            parameter_tuple = []
+            for k, v in parameters.items():
+                parameter_tuple.append((k, v))            
+            p = mp.Process(target=eval_in_parallel, args=(gpu_id, checkpoint, limit, parameter_tuple, return_dict, i, trial_idx))
+            p.start()
+            processes.append(p)
+
+        # Wait for all processes to finish
+        for p in processes:
+            p.join()
+
+        # Print the results after all processes have finished
+        print(return_dict)
+        for i in range(len(gpu_list)):
+            current_trial_id, config, eval_results = return_dict[i]
+            history.append((eval_results, config))
+            ax_client.complete_trial(trial_index=current_trial_id,raw_data=eval_results,)
+
+    print("------Finish BO------")
+    for h in history:
+        print(h)
+    print("------Best config------")
+    best_parameters, values = ax_client.get_best_parameters()
+    print(best_parameters, values)
+
+    print("------history------")
+    print(ax_client.generation_strategy.trials_as_df)
```

Comment:
How many trials do we usually get here? Probably makes sense to save this as a file somewhere too, and just print the summary statistics such as: best configs, num trials to get to best config, etc.

Reply:
Currently made the history part into a csv and print out the best config over all the history.